plan/preprocess.go

```diff
@@ -106,6 +113,10 @@ func (p *preprocessor) Leave(in ast.Node) (out ast.Node, ok bool) {
 		}
 	case *ast.TableName:
 		p.handleTableName(x)
+	case *ast.Join:
+		if len(p.tableAliasInJoin) > 0 {
+			p.tableAliasInJoin = p.tableAliasInJoin[:len(p.tableAliasInJoin)-1]
+		}
```

Comment:
Reset `isParentJoin` to `false`?

Reply:
`isParentJoin` works only in  function `Enter`, there's no need to clear it in `Leave`